File: FindMyHome.Frontend/src/app/models/listing/listing-query.model.ts
export class ListingQuery {
    public listingTypeIds: number[];
    public listingMarketingTypeIds: number[];
    public cityIds: number[];
    public countyIds: number[];
    public sortById: number;
    public searchTerm: string;

    public constructor(listingTypeIds: number[], listingCountyIds: number[], listingCityIds: number[], listingMarketingTypeIds: number[], sortById: number, searchTerm: string){
        this.listingTypeIds = listingTypeIds;
        this.listingMarketingTypeIds = listingMarketingTypeIds;
        this.cityIds = listingCityIds;
        this.countyIds = listingCountyIds;
        this.searchTerm = searchTerm;
        this.sortById = sortById;
    }
}

File: FindMyHome.Domain/DTOs/Listing/ListingQueryModel.cs
namespace FindMyHome.Domain.DTOs.Listing;

public class ListingQueryModel
{
    public IEnumerable<int> ListingTypeIds { get; set; }

    public IEnumerable<int> ListingMarketingTypeIds { get; set; }

    public IEnumerable<int> CityIds { get; set; }
    public IEnumerable<int> CountyIds { get; set; }

    public int? SortById { get; set; }

    public string SearchTerm { get; set; }
}
